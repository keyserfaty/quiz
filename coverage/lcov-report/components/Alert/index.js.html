<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Alert/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Alert</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { View, Text, ActivityIndicator } from 'react-native'
&nbsp;
import { styles } from './statics/styles'
&nbsp;
const Modal = (props) =&gt; {
  const { title, subtitle, spinner } = props
&nbsp;
  return (
    &lt;View style={styles.container}&gt;
      &lt;View style={styles.modal}&gt;
        &lt;View style={styles.titleContainer}&gt;
          { title &amp;&amp; (
            &lt;Text style={styles.title}&gt;
              {title}
            &lt;/Text&gt;
          ) }
          { subtitle &amp;&amp; (
            &lt;Text style={styles.subtitle}&gt;
              {subtitle}
            &lt;/Text&gt;
          ) }
          { spinner &amp;&amp; (
            &lt;View style={styles.spinner}&gt;
              &lt;ActivityIndicator size="large" color="#0000ff" /&gt;
            &lt;/View&gt;
          )}
        &lt;/View&gt;
        &lt;View style={styles.children}&gt;
          {props.children}
        &lt;/View&gt;
      &lt;/View&gt;
    &lt;/View&gt;
  )
}
&nbsp;
export default Modal
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 06:19:18 GMT-0500 (EST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
