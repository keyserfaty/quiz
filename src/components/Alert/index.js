import React from 'react'
import { View, Text, ActivityIndicator } from 'react-native'

import { styles } from './statics/styles'

const Modal = (props) => {
  const { type, title, subtitle, spinner } = props

  return (
    <View style={[styles.container, type]}>
      <View style={styles.modal}>
        <View style={styles.titleContainer}>
          { title && (
            <Text style={styles.title}>
              {title}
            </Text>
          ) }
          { subtitle && (
            <Text style={styles.subtitle}>
              {subtitle}
            </Text>
          ) }
          { spinner && (
            <View style={styles.spinner}>
              <ActivityIndicator size="large" color="#0000ff" />
            </View>
          )}
        </View>
        <View style={styles.children}>
          {props.children}
        </View>
      </View>
    </View>
  )
}

export default Modal
