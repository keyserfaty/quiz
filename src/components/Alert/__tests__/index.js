import React from 'react'

import Alert from '../index'

import Shallow from 'react-test-renderer/shallow'

const renderer = new Shallow()

test('Should render', () => {
  let component = renderer.render(<Alert msg='Some msg' />)

  expect(component).toMatchSnapshot('initial render')
})
