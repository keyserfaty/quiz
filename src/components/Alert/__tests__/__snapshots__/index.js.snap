// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render: initial render 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "height": 1334,
      "justifyContent": "flex-end",
      "position": "absolute",
      "width": 750,
    }
  }
>
  <Component
    style={
      Array [
        Object {
          "borderRadius": 3,
          "justifyContent": "space-between",
          "marginBottom": 8,
          "padding": 16,
          "width": "95%",
        },
        undefined,
      ]
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Component
        style={
          Object {
            "color": "rgba(0,0,0, 0.5)",
            "fontSize": 16,
            "lineHeight": 27,
          }
        }
      >
        Some msg
      </Component>
      <TouchableOpacity
        activeOpacity={0.2}
      >
        <Component
          style={
            Object {
              "color": "rgba(0,0,0, 0.5)",
              "fontSize": 16,
              "lineHeight": 27,
            }
          }
        >
          X
        </Component>
      </TouchableOpacity>
    </Component>
  </Component>
</Component>
`;
